### ISSUE TITLE
Problem with model initialization while running on GPU

### ISSUE THREAD

**Issue by supamee** [keras-team/keras#11078]
When I set up my model the loss is already at ~0. Because loss starts at 0 no amount of training helps. Is this a bug with the LTMS layer? This issue seems to only happen when using LTMS on the GPU. What's more, if I try to clear variables and re-initialize the models which ones start at loss 0 change(in the provided example the 1st and 3rd fail on the first initialization but upon re-initialization, only the 3rd fails)

    import os
    import numpy as np
    import tensorflow as tf
    from tensorflow import keras
   
    def gen_mem_data(amount,lookback,batch_size,block_size,high):
        if (amount/block_size)<batch_size:
            amount = batch_size * batch_size * 10
        Z = np.random.randint(0, high, amount+lookback)
        X=(Z[lookback:]-((high-1)/2))/((high-1)/2)
        Y=Z[:-lookback]
        if len(X)%block_size != 0:
            Y = Y[:-(len(Y) % block_size)]
            X = X[:-(len(X) % block_size)]
        X = X.reshape(int(len(X) / block_size), block_size)
        Y = Y.reshape(int(len(Y) / block_size), block_size)
        if len(X)%batch_size != 0:
            Y = Y[:-(len(Y) % batch_size)]
            X = X[:-(len(X) % batch_size)]
        X_lstm_train = X.reshape(X.shape[0], X.shape[1], 1)
        Y_lstm_train = keras.utils.to_categorical(Y)
        return X_lstm_train,Y_lstm_train,X,Y
    
    os.environ['TF_CPP_MIN_LOG_LEVEL'] = '3'
    config = tf.ConfigProto()
    config.gpu_options.allow_growth = True
    session = tf.Session(config=config)
    device_times = {
            "/gpu:0":[],
            "/cpu:0":[]
        }
    
    samples=100000
    lookback=2
    batch_size=100
    block_size=10
    epochs_per_run=1
    num_runs=25
    high=3
    _optimizer="Nadam"
    
    for device_name in device_times.keys():
        with tf.device(device_name):
            models = {}
            print("\nsetup:")
            X_lstm_train, Y_lstm_train, X, Y = gen_mem_data(amount=samples, lookback=lookback, batch_size=batch_size,
                                                            block_size=block_size, high=high)
            X_lstm_valid, Y_lstm_valid, _, _ = gen_mem_data(amount=int(samples / 10), lookback=lookback, batch_size=batch_size,
                                                            block_size=block_size, high=high)
    
            big_memory_lstm500 = model = keras.Sequential([
                keras.layers.LSTM(500, activation='tanh', input_shape=(X_lstm_train.shape[1], 1),
                                  return_sequences=True),
                keras.layers.Dense(10, activation='softmax'),
                keras.layers.Dense(Y_lstm_train.shape[2], activation='softmax')
            ])
            big_memory_lstm500.compile(optimizer=_optimizer, loss='categorical_crossentropy', metrics=['categorical_accuracy'])
            print("loss=",big_memory_lstm500.evaluate(X_lstm_valid, Y_lstm_valid, batch_size=batch_size, verbose=0)[0])
            models["big_memory_lstm500"] = big_memory_lstm500
    
            big_memory_lstm100 = keras.Sequential([
                keras.layers.LSTM(100, activation='tanh', input_shape=(X_lstm_train.shape[1], 1),
                                  return_sequences=True),
                keras.layers.Dense(10, activation='softmax'),
                keras.layers.Dense(Y_lstm_train.shape[2], activation='softmax')
            ])
            big_memory_lstm100.compile(optimizer=_optimizer, loss='categorical_crossentropy', metrics=['categorical_accuracy'])
            print("loss=",big_memory_lstm100.evaluate(X_lstm_valid, Y_lstm_valid, batch_size=batch_size, verbose=0)[0])
            models["big_memory_lstm100"] = big_memory_lstm100
    
            big_memory_lstm158 = keras.Sequential([
                    keras.layers.LSTM(158, activation = 'tanh', input_shape = (X_lstm_train.shape[1], 1),
                                                  return_sequences = True),
                    keras.layers.Dense(Y_lstm_train.shape[2], activation = 'softmax')
            ])
            big_memory_lstm158.compile(optimizer = _optimizer, loss = 'categorical_crossentropy', metrics=['categorical_accuracy'])
            print("loss=",big_memory_lstm158.evaluate(X_lstm_valid, Y_lstm_valid, batch_size=batch_size, verbose=0)[0])
            models["big_memory_lstm158"] = big_memory_lstm158
    
    
            print("start train")
            for i in range(num_runs):
                print("\ngen new train data")
                X_lstm_train, Y_lstm_train, X, Y = gen_mem_data(amount=samples, lookback=lookback, batch_size=batch_size,
                                                                block_size=block_size, high=high)
                for model in models:
                    models[model].fit(X_lstm_train,Y_lstm_train,
                                      epochs=epochs_per_run, batch_size=batch_size,
                                      verbose=0,validation_data=(X_lstm_valid, Y_lstm_valid))
                    eval = models[model].evaluate(X_lstm_valid, Y_lstm_valid, batch_size=batch_size, verbose=0)
                    print(model,i, "Loss: {:<5f}  Acc: {:<5f}".format(eval[0], eval[1]))
    
            print("end train")

Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [X] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps

- [X] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [X] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by jlherren** [comment]
What is the output that you obtain? Mine is as follows on TF 1.9.0 with a GTX 1060 which does not show any zero losses:

```
setup:
loss= 1.0986916542053222
loss= 1.1072739124298097
loss= 1.10063134431839
start train

gen new train data
big_memory_lstm500 0 Loss: 0.610000  Acc: 0.866200
big_memory_lstm100 0 Loss: 0.690896  Acc: 0.842400
big_memory_lstm158 0 Loss: 0.267786  Acc: 0.866100

gen new train data
big_memory_lstm500 1 Loss: 0.491698  Acc: 0.866700
big_memory_lstm100 1 Loss: 0.471218  Acc: 0.866700
big_memory_lstm158 1 Loss: 0.234083  Acc: 0.864500

gen new train data
big_memory_lstm500 2 Loss: 0.435431  Acc: 0.866200
big_memory_lstm100 2 Loss: 0.400268  Acc: 0.866700
big_memory_lstm158 2 Loss: 0.227572  Acc: 0.863700

...

gen new train data
big_memory_lstm500 24 Loss: 0.230875  Acc: 0.868000
big_memory_lstm100 24 Loss: 0.229116  Acc: 0.865000
big_memory_lstm158 24 Loss: 0.220584  Acc: 0.866100
end train
```
